### PR TITLE
lightning(dm): add config value to use local backend

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #3510

### What is changed and how it works?

`loaders.{config-name}.import-mode` can set "logical" or "physical", to use tidb or local backend of lightning respectively

#### Tests 

 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
ptal

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: dm/tests/lightning_mode/run.sh [hunk 1]
```diff
@@ -143,40 +63,15 @@ function run() {
 	check_log_not_contains $WORK_DIR/worker1/log/dm-worker.log "123456"
 	check_log_not_contains $WORK_DIR/worker2/log/dm-worker.log "123456"
 
-	# test drop table if exists
-	run_sql_source1 "drop table if exists \`lightning_mode\`.\`tb1\`;"
-	run_sql_source1 "drop table if exists \`lightning_mode\`.\`tb1\`;"
-	run_sql_source2 "drop table if exists \`lightning_mode\`.\`tb2\`;"
-	run_sql_source2 "drop table if exists \`lightning_mode\`.\`tb2\`;"
-	check_log_not_contains $WORK_DIR/worker1/log/dm-worker.log "Error .* Table .* doesn't exist"
-	check_log_not_contains $WORK_DIR/worker2/log/dm-worker.log "Error .* Table .* doesn't exist"
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/tests/lightning_mode/run.sh:152]: seems these tests are duplicated with all_mode, I guess the original developer copied from there. Many tests are not needed to repeat again

File: dm/tests/lightning_mode/run.sh [hunk 2]
```diff
@@ -143,40 +63,15 @@ function run() {
...
-	export GO_FAILPOINTS=''
-
 	run_sql_both_source "SET @@GLOBAL.SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION'"
+
+	# restart to standalone tidb
+	killall -9 tidb-server 2>/dev/null || true
+	killall -9 tikv-server 2>/dev/null || true
+	killall -9 pd-server 2>/dev/null || true
+	rm -rf /tmp/tidb || true
+	rm -rf /tmp/1000_tidb || true
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/tests/lightning_mode/run.sh:73]: we started a real TiDB cluster (with PD/TiKV) in this case, after this case I want to recover the old unistore TiDB to continue on other cases, so clean the temporary files and restart TiDB

[the remark this answers is not in the document]
Comment (lichunzhu) [reply dm/tests/lightning_mode/run.sh:73]: Where is `1000_tidb` generated in test 🤔 ?

[the remark this answers is not in the document]
Comment (lance6716) [reply dm/tests/lightning_mode/run.sh:73]: seems this is generated by tiup, I'll remove it soon

**Review by lichunzhu — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
ptal @GMHDBJD @buchuitoudegou

**Review by buchuitoudegou — APPROVED** [review]
lgtm

**Pull request merged**